Adapter.C

```diff
@@ -831,411 +831,167 @@ void preciceAdapter::Adapter::setupMeshVolCheckpointing()
 
 void preciceAdapter::Adapter::setupCheckpointing()
 {
-    // Add fields in the checkpointing list
-    DEBUG(adapterInfo("Creating a list of checkpointed fields..."));
+    // Add fields in the checkpointing list - sorted for parallel consistency
+    DEBUG(adapterInfo("Adding in checkpointed fields..."));
 
-    /* Find and add all the registered objects in the mesh_
-       of type volScalarField
-    */
-
-    // Print the available objects of type volScalarField
-    DEBUG(adapterInfo("Collecting objects of type volScalarField... "));
-
-    wordList objectNames_ = mesh_.lookupClass<volScalarField>().toc();
-
-    forAll(objectNames_, i)
-    {
-        if (mesh_.foundObject<volScalarField>(objectNames_[i]))
-        {
-            addCheckpointField(
-                const_cast<volScalarField&>(
-                    mesh_.lookupObject<volScalarField>(objectNames_[i])));
-
-#ifdef ADAPTER_DEBUG_MODE
-            adapterInfo(
-                "Will be checkpointing " + objectNames_[i]);
-#endif
-        }
-        else
-        {
-            adapterInfo("Could not checkpoint " + objectNames_[i], "warning");
-        }
+#undef doLocalCode
+#define doLocalCode(GeomField)                                           \
+    /* Checkpoint registered GeomField objects */                        \
+    for (const word& obj : mesh_.sortedNames<GeomField>())               \
+    {                                                                    \
+        addCheckpointField(mesh_.thisDb().getObjectPtr<GeomField>(obj)); \
+        DEBUG(adapterInfo("Checkpoint " + obj + " : " #GeomField));      \
```

Comment:
Without a request to change this: Is there a specific reason that you prefer a macro over a template?

Reply:
This indeed could be rewritten as a (static) templated private method taking three arguments (mesh db, vector of pointers, vector of pointers for the copy).
Could also unroll the methods and online them within the macro.
The main reason for the macro approach is to minimise the changes to the code and also leave the individual methods as-is in case there is a need to do something more than the push_back that it currently does. Or if surface fields need different treatment etc.

Reply:
upper or lowercase doesn't really matter. In recent times I've abandoned proper names and just gone with 'doLocalCode', which describes exactly what it is. The advantage of this one is that we have have the undef/define/undef all in local scope and don't have to ever worry about an accidental undef on a meaningful macro name.

Templates are also fine, but they would add a file-scope location function.  In this case however, you may still want to wrap with a macro.  There are probably other places to fix first.

Reply:
I would keep it as suggested here, considering that the same coding style is used throughout the OpenFOAM code base in order to keep things consistent.

Reply:
Then I also fully agree. The name was just not very clear to me on the first sight.